ui/components/Link.tsx

```diff
@@ -3,6 +3,14 @@ import { Link as RouterLink } from "react-router-dom";
 import styled from "styled-components";
 import Text, { TextProps } from "./Text";
 
+function isAllowed(uri: string): boolean {
+  // Regex from https://github.com/cure53/DOMPurify/blob/cce00ac40d33c2aae6422eaa59e6a8aad5c73901/src/regexp.js
+  const regex = new RegExp(
+    /^(?:(?:(?:f|ht)tps?|mailto|tel|callto|cid|xmpp):|[^a-z]|[a-z+.\-]+(?:[^a-z+.\-:]|$))/i // eslint-disable-line no-useless-escape
+  );
+  return regex.test(uri);
+}
+
```

Comment:
Is it possible/worth it to replace `isHTTP` from lib/utils line 30 with this? We're definitely using it in Metadata to see if we should render a link or text, but we don't need to do that anymore as this component now returns text if the link is invalid. It'd be nice to not have to run what's basically the same function twice - it's not a big deal since it's literally one regex but yeah.

Reply:
This regex allows relative links, so if it was the same function then `/foo` would become a link, which I don't _think_ we want.

Reply:
Moved it to `utils` and wrote some comments to explain what the difference is.